source/Halibut/ServiceModel/PendingRequestQueueAsync.cs

```diff
@@ -150,11 +158,12 @@ public async Task WaitUntilComplete(CancellationToken cancellationToken)
                 catch (Exception ex) when (ex is TaskCanceledException or OperationCanceledException)
                 {
                     // responseWaiter.Set is only called when the request has been collected and the response received.
-                    // It is possible that the transfer has already started once the cancellationToken is cancelled
-                    // In this case we cannot walk away from the request as it is already in progress and no longer in the connecting phase
+                    // It is possible that the transfer has already started once the requestCancellationTokens.LinkedCancellationToke is cancelled
+                    // If the requestCancellationTokens.InProgressCancellationToken is Ct.None or not cancelled then
+                    // we cannot walk away from the request as it is already in progress and no longer in the connecting phase
                     cancelled = true;
                     
-                    using (await transferLock.LockAsync())
+                    using (await transferLock.LockAsync(CancellationToken.None))
```

Comment:
Using none is a good idea, it makes it easier to see places that perhaps need a different token.